internal/sysfs/open_file_windows.go

```diff
@@ -154,46 +148,3 @@ func open(path string, mode int, perm uint32) (fd syscall.Handle, err error) {
 	h, e := syscall.CreateFile(pathp, access, sharemode, sa, createmode, attrs, 0)
 	return h, e
 }
-
-// windowsOsFile overrides osFile to special case directory handling in Windows.
-type windowsOsFile struct {
-	osFile
-
-	dirInitialized bool
-}
-
-// Readdir implements File.Readdir
-func (f *windowsOsFile) Readdir(n int) (dirents []fsapi.Dirent, errno syscall.Errno) {
-	if errno = f.maybeInitDir(); errno != 0 {
-		return
-	}
-
-	return f.osFile.Readdir(n)
-}
-
-func (f *windowsOsFile) maybeInitDir() syscall.Errno {
-	if f.dirInitialized {
-		return 0
-	}
-
-	if isDir, errno := f.IsDir(); errno != 0 {
-		return errno
-	} else if !isDir {
-		return syscall.ENOTDIR
-	}
-
-	// On Windows, once the directory is opened, changes to the directory are
-	// not visible on ReadDir on that already-opened file handle.
-	//
-	// To provide consistent behavior with other platforms, we re-open it.
-	if errno := f.osFile.Close(); errno != 0 {
```

Comment:
@achille-roussel so this PR is moving this code and conditional to where it happens under an if statement instead of a subtype. So, really the change should be cosmetic, not actually behavioral. OTOH, it is good advice to contain this to windows, and I think we can!

Reply:
I crossed context with discussions on Slack and misunderstood the goal of this change.

I'm glad there's no increase to the risk surface, and the scope of reopening on directory reads only on Windows is a more comfortable trade-off 👍

Reply:
ps your feedback dramatically affects #1529 which was on the path to always re-open. I'll make sure we don't do that, and probably it is better to just make changes like this so each part are visible.